R/utils.R

```diff
@@ -454,3 +454,50 @@ replace_or_append <- function(lines, pattern, replacement) {
   }
   lines
 }
+
+# Request an OAuth access token for the given resource from Posit Connect. The
+# OAuth token will belong to the user owning the given Shiny session.
+connect_viewer_token <- function(session, resource) {
+  # Ensure we're running on Connect.
+  server_url <- Sys.getenv("CONNECT_SERVER")
+  api_key <- Sys.getenv("CONNECT_API_KEY")
+  if (nchar(server_url) == 0 || nchar(api_key) == 0) {
+    return(NULL)
+  }
+
+  # Ensure we have a Shiny session object.
+  if (is.null(session) || is.null(session$request)) {
+    return(NULL)
+  }
+
+  # Older versions or certain configurations of Connect might not supply a user
+  # session token.
+  token <- session$request$HTTP_POSIT_CONNECT_USER_SESSION_TOKEN
+  if (is.null(token)) {
+    return(NULL)
+  }
+
+  # See: https://docs.posit.co/connect/api/#post-/v1/oauth/integrations/credentials
+  req <- httr2::request(server_url)
+  req <- httr2::req_url_path_append(
+    req, "__api__/v1/oauth/integrations/credentials"
+  )
+  req <- httr2::req_headers(req,
+    Authorization = paste("Key", api_key), .redact = "Authorization"
```

Comment:
This is a custom auth protocol, right? Not a standard. (i.e. it's fine that httr2 doesn't have an existing helper for this)

Reply:
With `Key` and not `Bearer` as a prefix?